[flake]: TestBatchedAssetIssuance/fund_seal_on_restart flake

This tapgarden case was observed flaking [in CI](https://github.com/lightninglabs/taproot-assets/actions/runs/22774350401/job/66063643191?pr=2017).

```
--- FAIL: TestBatchedAssetIssuance (57.34s)
    --- FAIL: TestBatchedAssetIssuance/fund_seal_on_restart (10.11s)
        test_sqlite.go:11: Creating new SQLite DB handle for testing: /tmp/TestBatchedAssetIssuancefund_seal_on_restart4099760947/001/tmp.db
        planter_test.go:2058: 
            	Error Trace:	/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:402
            	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:2058
            	            				/home/runner/work/taproot-assets/taproot-assets/tapgarden/planter_test.go:2123
            	Error:      	Should be zero, but was 1
            	Test:       	TestBatchedAssetIssuance/fund_seal_on_restart
```

--- a/tapgarden/planter.go
+++ b/tapgarden/planter.go
@@ -610,6 +610,18 @@ func (c *ChainPlanter) Start() error {
 				// can now be set as frozen. We are already not
 				// able to add new seedlings to the batch.
 				batch.UpdateState(BatchStateFrozen)
+
+				err := c.cfg.Log.UpdateBatchState(
+					ctx, batch.BatchKey.PubKey,
+					BatchStateFrozen,
+				)
+				if err != nil {
+					log.Warnf("Failed to update batch "+
+						"state to frozen (%x): %s",
+						batchKey, err.Error())
+					cancelBatch()
+					continue
+				}
 			}
 
 			log.Infof("Launching ChainCaretaker(%x)", batchKey)

